tensorflow/compiler/xla/service/gpu/ir_emitter.cc

```diff
@@ -289,11 +291,51 @@ bool IrEmitter::MaybeEmitDirectAtomicOperation(
       }
     }
 
+    if (isAMDGPU) {
+      std::string arch = ir_emitter_context_->amdgpu_arch();
+      llvm::PointerType* output_address_type =
+          llvm::dyn_cast<llvm::PointerType>(output_address->getType());
```

Comment:
`auto` preferred when type is duplicated

Reply:
fixed